test/unit/commands/actions-specs.js

```diff
@@ -48,4 +50,41 @@ describe('Actions', () => {
       driver.bootstrap.sendAction.calledWith('drag').should.be.true;
     });
   });
+  describe('pullFolder', () => {
+    it('should pull a folder and return base64 zip', async () => {
+
+      let adbPullStub;
+      const pull = async (ignore, localPath) => {
+        await mkdirp(path.resolve(localPath));
+        await fs.writeFile(path.resolve(localPath, 'a.txt'), 'hello world', {flags: 'w'});
+        await fs.writeFile(path.resolve(localPath, 'b.txt'), 'foobar', {flags: 'w'});
+      };
+
+      // Stub in driver.adb
+      if (!driver.adb) {
+        driver.adb = {pull};
+      } else {
+        adbPullStub = sinon.stub(driver.adb, 'pull', pull);
+      }
+
+      // Get the zipped buffer contents and write it to a .zip file
+      const buffer = await driver.pullFolder('/does/not/matter');
+      (typeof(buffer)).should.equal('string');
+      const zippedDir = await tempDir.openDir();
+      await fs.writeFile(path.resolve(zippedDir, 'zipped.zip'), buffer, 'base64');
+      
+      // Extract the zip file
+      const unzippedDir = await tempDir.openDir();
```

Comment:
I found a library called `mock-fs` that overrides `fs` and does everything in memory. I think I'll use that as a dev-dependency.